openassessment/workflow/test/test_api.py

```diff
@@ -254,6 +255,39 @@ def test_unexpected_peer_workflow_errors_wrapped(self, mock_create):
         submission = sub_api.create_submission(ITEM_1, ANSWER_2)
         workflow_api.create_workflow(submission["uuid"], ["peer", "self"], ON_INIT_PARAMS)
 
+    @patch.object(staff_api, 'get_score')
+    @patch.object(PeerWorkflow.objects, 'get')
+    def test_no_peer_assessment_error_handled(self, mock_get_workflow, mock_get_staff_score):
+        """
+        Tests to verify that, given a problem that requires the peer step and a submission associated with a workflow
+        that has no assessments, an overriding staff score will push the workflow into the done state and not crash
+        when there are no assessments in the "completed" peer step.
+        """
+        mock_get_workflow.return_value = None
+        mock_get_staff_score.return_value = {
+            "points_earned": 10,
+            "points_possible": 10,
+            "contributing_assessments": 123,
+            "staff_id": "staff 1",
+        }
+        _, submission = self._create_workflow_with_status(
+            "user 1",
+            "test/1/1",
+            "peer-problem",
+            "peer",
+            steps=["peer"]
+        )
+        workflow_api.update_from_assessments(
```

Comment:
This is the entry point for the crashed-previously codepath. The key difference from prior behavior is that it's now possible to have a "completed" peer step without any peer workflows existing.